src/oncall/scheduler/default.py

```diff
@@ -501,12 +447,20 @@ def populate(self, schedule, start_time, dbinfo):
         future_events = filter(lambda x: x != [], future_events)
         if future_events:
             first_event_start = min(future_events[0], key=lambda x: x['start'])['start']
-            cursor.execute('DELETE FROM event WHERE schedule_id = %s AND start >= %s', (schedule['id'], first_event_start))
+            query = 'DELETE FROM %s WHERE schedule_id = %s AND start >= %s' % (table_name, schedule['id'], first_event_start)
+            cursor.execute(query)
 
         # Create events in the db, associating a user to them
         for epoch in future_events:
-            user_id = self.find_next_user_id(schedule, epoch, cursor)
+            user_id = self.find_next_user_id(schedule, epoch, cursor, table_name)
             if not user_id:
                 continue
-            self.create_events(team_id, schedule['id'], user_id, epoch, role_id, cursor)
-        connection.commit()
\ No newline at end of file
+            self.create_events(team_id, schedule['id'], user_id, epoch, role_id, cursor, table_name)
+
+        # if this function is being called by preview return events
+        if req.method == 'GET':
+            resp.body = self.build_preview_response(req, resp, cursor, table_name)
+            query = "DROP TABLE `temp_event`"
```

Comment:
This seems to ignore your custom `table_name` variable. The change i had in mind was:

    query = "DROP TABLE %s"
    cursor.execute(query, [table_name])

Reply:
It does indeed ignore it. Revisiting it, it didn't seem necessary to use the variable since only preview uses GET and preview always uses the table name 'temp_event'. In fact, 'temp_event' is already hardcoded within the columns and constraints. This way seemed bit more readable and unlikely to drop the original 'event' database if for some reason a GET endpoint for populate was added.

Reply:
ah gotcha. should probably remove all references to the `table_name` variable throughout the code/PR for consistency if it won't be used

Reply:
So table_name is used when creating events and deleting events (since both preview and populate use the same functions), it's just the creating and dropping of the temporary table that  don't need a variable since they are unique to preview.

Reply:
ah that makes sense